joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala

```diff
@@ -311,10 +323,16 @@ trait KtPsiToAst {
     )
     scope.pushNewScope(_methodNode)
 
-    val parameters = withIndex(ktFn.getValueParameters.asScala.toSeq) { (p, idx) =>
-      astForParameter(p, idx)
-    }.flatMap(_.root.collectAll[NewMethodParameterIn])
+    val thisParameterMaybe = if (needsThisParameter) {
+      val typeDeclFullName = registerType(typeInfoProvider.containingTypeDeclFullName(ktFn, TypeConstants.any))
+      val node             = methodParameterNode(Constants.this_, typeDeclFullName, line(ktFn), column(ktFn)).order(0)
+      scope.addToScope(Constants.this_, node)
+      Some(node)
+    } else None
```

Comment:
You can rewrite this as
```
Option.when(needsThisParameter) {
  val typeDeclFullName ...
  ...
}
```
which is IMO cleaner than `if (something) Some(thing) else None`

Reply:
not a fan on a first look, will leave the decision of using _Option.when_ for later